options/path.c

```diff
@@ -203,9 +203,10 @@ char *mp_get_user_path(void *talloc_ctx, struct mp_log *log,
         // parse to "~" <prefix> "/" <rest>
         bstr prefix, rest;
         if (bstr_split_tok(path, "/", &prefix, &rest)) {
-            const char *rest0 = rest.start; // ok in this case
             if (bstr_equals0(prefix, "~")) {
+                char *rest0 = bstrdup0(NULL, rest);
```

Comment:
This commit should be squashed into `options/path: change mp_get_user_path to accept bstr directly`.

You made this usage invalid, when passed string was null-terminated, it was fine to use rest.start.

Reply:
Is that right? I was thought that rest.start was never guaranteed to be safe to use but it just happened to be because everything that called this was internal and had null-terminated strings. What made things break, as far as I can tell, was trying to call this in the parser. Everything else seemed OK.

But I don't mind squashing it if you want.